crates/proto/src/tcp/mod.rs

```diff
@@ -26,10 +26,24 @@ pub use self::tcp_stream::{Connect, DnsTcpStream, TcpStream};
 pub mod tokio {
     use std::io;
     use std::net::SocketAddr;
+    use tokio::net::TcpSocket as TokioTcpSocket;
     use tokio::net::TcpStream as TokioTcpStream;
 
-    pub async fn connect(addr: &SocketAddr) -> Result<TokioTcpStream, io::Error> {
-        let stream = TokioTcpStream::connect(addr).await?;
+    pub async fn connect(
```

Comment:
I think we should create a new method, `connect_with_bind` (or other better name) and leave the pub interface the same as I don't think we need to cause a breaking change for this, and most people may not need this.

Reply:
Done